test/DefifaGovernor.t.sol

```diff
@@ -160,42 +164,503 @@ contract DefifaGovernorTest is TestBaseWorkflow {
+    //             1 ether,
+    //             address(0),
+    //             _users[i],
+    //             0,
+    //             true,
+    //             "",
+    //             metadata
+    //         );
+
+    //         // Set the delegate as the user themselves
+    //         vm.prank(_users[i]);
+    //         _nft.setTierDelegate(_users[i], uint256(i + 1));
+
+    //         // Forward 1 block, user should receive all the voting power of the tier, as its the only NFT
+    //         vm.roll(block.number + 1);
+    //         assertEq(
+    //             _governor.MAX_VOTING_POWER_TIER(),
+    //             _governor.getVotes(_users[i], block.number - 1)
+    //         );
+    //     }
+
+    //     address[] memory targets = new address[](1);
+    //     uint256[] memory values = new uint256[](1);
+    //     bytes[] memory calldatas = new bytes[](1);
+
+    //     targets[0] = address(_nft);
+    //     calldatas[0] = abi.encodeCall(
+    //         _nft.setTierRedemptionWeights,
+    //         scorecards
+    //     );
+
+    //     // Forward time so proposals can be created
+    //     vm.warp(block.timestamp + _governor.proposalCreationThreshold() + 1);
+
+    //     // Create the proposal
+    //     uint256 _proposalId = _governor.propose(
+    //         targets,
+    //         values,
+    //         calldatas,
+    //         "Governance!"
+    //     );
+
+    //     // Forward time so voting becomes active
+    //     vm.roll(block.number + _governor.votingDelay() + 1);
+
+    //     // All the users vote
+    //     // 0 = Against
+    //     // 1 = For
+    //     // 2 = Abstain
+    //     for (uint256 i = 0; i < _users.length; i++) {
+    //         vm.prank(_users[i]);
+    //         _governor.castVote(_proposalId, 1);
+    //     }
+
+    //     // Forward time to the block after voting closes
+    //     vm.roll(_governor.proposalDeadline(_proposalId) + 1);
+
+    //     // uint256 _cumulativeRedemptionWeight;
+    //     // for (uint256 i = 0; i < scorecards.length; i++) {
+    //     //     _cumulativeRedemptionWeight += scorecards[i].redemptionWeight;
+    //     // }
+    //     // if (_cumulativeRedemptionWeight > 1_000_000_000) {
+    //     //     vm.expectRevert(
+    //     //         abi.encodeWithSignature("INVALID_REDEMPTION_WEIGHTS()")
+    //     //     );
+    //     // }
+
+    //     // // Execute the proposal
+    //     _governor.execute(targets, values, calldatas, keccak256("Governance!"));
+
+    //     // Verify that the redemptionWeights actually changed
+    //     for (uint256 i = 0; i < scorecards.length; i++) {
+    //         // since there can be a case of duplicate id's hence modified this check & to make sure we have enough inputs the redemption weight can be 0 as well
+    //         if (scorecards[i].redemptionWeight != 0) {
+    //             assert(_nft.tierRedemptionWeights(scorecards[i].id) >= 0);
+    //         }
+    //     }
+    // }
+
```

Comment:
Lots of commented code, should this be uncommented or removed?

Reply:
hey just seeing this now, yh commented right now because of the low input error from forge so will get into this later let's leave this commented for now